### PR TITLE
Enable IdentityBinaryExpression error-prone check

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

File: core/trino-spi/src/main/java/io/trino/spi/function/ScalarFunctionAdapter.java [hunk 1]
```diff
@@ -190,7 +190,7 @@ public MethodHandle adapt(
         requireNonNull(methodHandle, "methodHandle is null");
         requireNonNull(actualConvention, "actualConvention is null");
         requireNonNull(expectedConvention, "expectedConvention is null");
-        if (expectedConvention.getArgumentConventions().size() != expectedConvention.getArgumentConventions().size()) {
+        if (actualConvention.getArgumentConventions().size() != expectedConvention.getArgumentConventions().size()) {
```
Comment (wendigo) [inline core/trino-spi/src/main/java/io/trino/spi/function/ScalarFunctionAdapter.java:193]: That one is actually interesting

File: core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ public void testSubtract()
         assertFunction("100000000037 - 37", BIGINT, 100000000037L - 37L);
         assertFunction("37 - 100000000017", BIGINT, 37 - 100000000017L);
         assertFunction("100000000017 - 37", BIGINT, 100000000017L - 37L);
-        assertFunction("100000000017 - 100000000017", BIGINT, 100000000017L - 100000000017L);
+        assertFunction("100000000017 - 100000000017", BIGINT, 0L);
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java:74]: That's a false positive in a sense that the original code was intentionally written this way.
I don't think it's a big issue, but worth noting.

**Review by findepi — APPROVED** [review]
LGTM but would like hear @martint thoughts if he finds time.

File: core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java (hunk 2 above)
Comment (kokosing) [reply core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java:74]: We can have this as a comment?

Comment (ksobolew) [reply core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java:74]: In my series adding Error Prone checks, I just added a suppression for this, as I recognized that this was intentional:
https://github.com/trinodb/trino/pull/6879/files#diff-2fec98bebd272276c2c0cafe53035760f5a45e6cc80f16be071aeecac6dcbf76R69

Comment (findepi) [reply core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java:74]: Is it a question to me?

Comment (ksobolew) [reply core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java:74]: (The commit from the link above is part of a longer series which I started to split into smaller pieces, I haven't gotten to this one yet)

Comment (kokosing) [reply core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java:74]: > Is it a question to me?

To audience (author and other reviewers).

Comment (findepi) [reply core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java:74]: @ksobolew suppressions are a signal that check may not be helpful overall.

Comment (wendigo) [reply core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java:74]: Yes, I agree but this change doesn't make code less readable imo

Comment (ksobolew) [reply core/trino-main/src/test/java/io/trino/type/TestBigintOperators.java:74]: @findepi yeah, I agree, but there are always exceptions and oddball cases, like this one. Ideally there should be a comment why the suppression is warranted, though, and I acknowledge that I didn't do that in my commit

**Review by ksobolew — APPROVED** [review]
It's unfortunate to break the pleasant symmetry of the expressions, but it's not a didaster either, so if others approve, I approve as well :)

File: core/trino-spi/src/main/java/io/trino/spi/function/ScalarFunctionAdapter.java (hunk 1 above)
Comment (findepi) [reply core/trino-spi/src/main/java/io/trino/spi/function/ScalarFunctionAdapter.java:193]: I assume this currently has no end-user visible impact and i won't write release notes for this.
@dain  please correct me if I am wrong here.

**Pull request merged**